render/fx_renderer/gles2/shaders/box_shadow.frag

```diff
@@ -80,7 +87,10 @@ void main() {
     shadow_alpha += (random() - 0.5) / 128.0;
 
     // get the window alpha so we can render around the window (fix pixel gap by adding 1.0 to radius)
-    float window_alpha = smoothstep(-1.0, 1.0, roundRectSDF(window_half_size, window_position, window_corner_radius + 1.0));
+    float window_corner_alpha = hole_alpha(window_size, window_position,
+            window_corner_radius,
+            window_round_top_left, window_round_top_right,
+            window_round_bottom_left, window_round_bottom_right);
```

Comment:
why not reuse the corner_alpha function? why do we need a new hole_alpha shader?

Reply:
I realize I fixed up that function after this PR was created, but I'd think we can just reuse it here like we had roundRectSDF

Reply:
The reason why I created the new function is due to the failsafes that the hole needs. Without it, corners set to none while a corner radius is set would create a wonky output.

But I could copy/paste that logic into the rounded quad shader as well

Reply:
I think we should safeguard both shaders in that case, they should be the same